packit/api.py

```diff
@@ -454,8 +454,13 @@ def create_srpm(
 
         current_git_describe_version = self.up.get_current_version()
         upstream_ref = upstream_ref or self.package_config.upstream_ref
-        commit = self.up.local_project.git_repo.active_branch.commit.hexsha[:8]
-
+        try:
+            commit = self.up.local_project.git_repo.active_branch.commit.hexsha[:8]
+        except TypeError as e:
+            if "detached symbolic reference" in str(e):
+                commit = str(e).split("'")[-2][:8]
```

Comment:
isn't there a better way to get it via the gitpython interface `self.up.local_project.git_repo`?

parsing it from the error message looks really ugly

Reply:
Something like `self.up.local_project.git_repo.head.is_detached` should do the trick.